src/typing/typer.ml

```diff
@@ -281,6 +281,20 @@ let rec type_ident_raise ctx i p mode with_type =
 			AKExpr (get_this ctx p)
 		| (MCall _, KAbstractImpl _) | (MGet, _)-> AKExpr(get_this ctx p)
 		| _ -> AKNo i)
+	| "abstract" ->
+		begin match mode, ctx.curclass.cl_kind with
+			| MSet _, KAbstractImpl ab -> typing_error "Property 'abstract' is read-only" p;
+			| (MGet, KAbstractImpl ab)
+			| (MCall _, KAbstractImpl ab) ->
+				let t = TAbstract (ab,[]) in
+				let vtmp = alloc_var VGenerated "tmp" t p in
```

Comment:
Why not just `(cast this:AbstractType)` instead of generating a temp var?

Reply:
Right... Seems like simple typed TBlock works, not sure if unsafe cast is required?